### ISSUE TITLE
need  sampled softmax loss  function to train model to get user vector and item vector with a large dictionary

### ISSUE THREAD

**Issue by zhangqianjin** [pytorch/pytorch#58351]
## 🚀 Feature
refer to https://www.tensorflow.org/api_docs/python/tf/nn/sampled_softmax_loss

## Motivation
 I want to use the trained model result for model recall. The existing torch.nn.AdaptiveLogSoftmaxWithLoss does not return the nearest neighbor result. And pytorch lacks the sampled softmax loss function (refer to https://www.tensorflow.org/api_docs/python/tf/nn/sampled_softmax_loss), which is very inconvenient for us to recall based on the model vector results（user vector and item vector）. The sampled_softmax_loss function is the my largest motivation to use tensorflow, especially eager for pytorch to add this function
